src/main/java/frc/robot/subsystems/climber/ClimberIOTalonFX.java

```diff
@@ -0,0 +1,225 @@
+import com.ctre.phoenix6.hardware.core.CoreCANcoder;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import frc.lib.team3015.subsystem.FaultReporter;
+import frc.lib.team3061.RobotConfig;
+import frc.lib.team3061.drivetrain.swerve.Conversions;
+import frc.lib.team6328.util.Alert;
+import frc.lib.team6328.util.Alert.AlertType;
+import frc.lib.team6328.util.TunableNumber;
+
+/** TalonFX implementation of the generic SubsystemIO */
+public class ClimberIOTalonFX implements ClimberIO {
+  private TalonFX leftMotor;
+  private TalonFX rightMotor;
+
+  private CoreCANcoder leftEncoder;
+  private CoreCANcoder rightEncoder;
+
+  private MotionMagicExpoTorqueCurrentFOC leftPositionCurrentRequest;
+  private TorqueCurrentFOC leftSupplyAmpsRequest;
+
+  private MotionMagicExpoTorqueCurrentFOC rightPositionCurrentRequest;
+  private TorqueCurrentFOC rightSupplyAmpsRequest;
+
+  private StatusSignal<Double> leftVelocityRPSStatusSignal;
+  private StatusSignal<Double> leftPositionRotationsStatusSignal;
+  private StatusSignal<Double> leftStatorCurrentAmpsStatusSignal;
+  private StatusSignal<Double> leftSetpointStatusSignal;
+  private StatusSignal<Double> leftSupplyCurrentAmpsStatusSignal;
+
+  private StatusSignal<Double> rightVelocityRPSStatusSignal;
+  private StatusSignal<Double> rightPositionMetersStatusSignal;
+  private StatusSignal<Double> rightStatorCurrentAmpsStatusSignal;
+  private StatusSignal<Double> rightSetpointStatusSignal;
+  private StatusSignal<Double> rightSupplyCurrentAmpsStatusSignal;
+
+  private Alert configAlert =
+      new Alert("Failed to apply configuration for Climber.", AlertType.ERROR);
+
+  // FIXME: Add KV, KA,
+  private final TunableNumber kP = new TunableNumber("Climber/KP", KP);
+  private final TunableNumber kI = new TunableNumber("Climber/KI", KI);
+  private final TunableNumber kD = new TunableNumber("Climber/KD", KD);
+
+  private final TunableNumber kS = new TunableNumber("Climber/KS", KS);
+  private final TunableNumber kV = new TunableNumber("Climber/KV", KV);
+  private final TunableNumber kA = new TunableNumber("Climber/KA", KA);
+  private final TunableNumber kVExpo = new TunableNumber("Climber/KVExpo", KV_EXPO);
+  private final TunableNumber kAExpo = new TunableNumber("Climber/KAExpo", KA_EXPO);
+
+  /** Create a TalonFX-specific generic SubsystemIO */
+  public ClimberIOTalonFX() {
+    configMotors(leftMotor, rightMotor);
+
+    leftVelocityRPSStatusSignal = leftMotor.getVelocity();
+    leftPositionRotationsStatusSignal = leftMotor.getPosition();
+    leftStatorCurrentAmpsStatusSignal = leftMotor.getStatorCurrent();
+    leftSetpointStatusSignal = leftMotor.getClosedLoopReference();
+
+    rightVelocityRPSStatusSignal = rightMotor.getVelocity();
+    rightPositionMetersStatusSignal = rightMotor.getPosition();
+    rightStatorCurrentAmpsStatusSignal = rightMotor.getStatorCurrent();
+    rightSetpointStatusSignal = rightMotor.getClosedLoopReference();
```

Comment:
also initialize rightSupplyCurrentAmpsStatusSignal and leftSupplyCurrentAmpsStatusSignal